### PR TITLE
Write test for dropdown component

### PR DESCRIPTION
**Pull request by anishpawaskar**
Date: `26-11-2023`

Developer Name: @anishpawaskar 

----

## Issue Ticket Number

-   Closes: https://github.com/Real-Dev-Squad/members-site/issues/104

## Description
- Write test for dropdown component
- Write test for logout api

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Under Feature Flag**

-   [ ] Yes
-   [x] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

## Test Coverage
image

Logout API test :-
image

Provide information on the test coverage for the changes made in this PR. Include details on the types of tests conducted and their outcomes.

### CODE REVIEW

File: src/mocks/handlers/logout.handler.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { rest } from "msw";
+import { LOGOUT_API } from "../../components/Dropdown/DropdownConstants";
+
+
+const logoutHandler = [
+    rest.get(`${LOGOUT_API}`, (_, res, ctx) => {
+        return res(
+            ctx.status(200),
+            ctx.json({
+                message: 'Signout successful',
```
Comment (sahsisunny) [inline src/mocks/handlers/logout.handler.ts:10]: Is this the message returned by the API?

File: src/test/unit/components/dropdown/Dropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        )
+
+        const signoutButton = screen.getByTestId("signoutButton");
+        expect(signoutButton).toBeInTheDocument();
+
+        fireEvent.click(signoutButton);
+
+        await waitFor(() => {
+            expect(mockLogoutUser).toHaveBeenCalled();
+            // Add additional assertions for success or error handling
```
Comment (sahsisunny) [inline src/test/unit/components/dropdown/Dropdown.test.tsx:43]: Is it necessary?

File: src/test/unit/components/dropdown/Dropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        )
+
+        const signoutButton = screen.getByTestId("signoutButton");
+        expect(signoutButton).toBeInTheDocument();
+
+        fireEvent.click(signoutButton);
+
+        await waitFor(() => {
+            expect(mockLogoutUserFailure).toHaveBeenCalled();
+            // Add additional assertions for success or error handling
```
Comment (sahsisunny) [inline src/test/unit/components/dropdown/Dropdown.test.tsx:65]: ?

File: src/mocks/handlers/logout.handler.ts (hunk 1 above)
Comment (anishpawaskar) [reply src/mocks/handlers/logout.handler.ts:10]: Yes

File: src/test/unit/components/dropdown/Dropdown.test.tsx (hunk 2 above)
Comment (anishpawaskar) [reply src/test/unit/components/dropdown/Dropdown.test.tsx:43]: Yes it is used to mock the logout function being called or not will remove comments

File: src/test/hooks/logoutApi.test.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+beforeAll(() => {
+  server.listen();
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+function Wrapper({
+  children,
+}: PropsWithChildren<Record<string, any>>): JSX.Element {
```
Comment (sahsisunny) [inline src/test/hooks/logoutApi.test.tsx:22]:
```suggestion
}: { children: React.ReactNode }): JSX.Element {
```

File: src/mocks/handlers/logout.handler.ts (hunk 1 above)
Comment (sahsisunny) [reply src/mocks/handlers/logout.handler.ts:10]: Can you share the API or screenshot?

File: src/test/unit/components/dropdown/Dropdown.test.tsx (hunk 2 above)
Comment (sahsisunny) [reply src/test/unit/components/dropdown/Dropdown.test.tsx]: I was asking about comments.

File: src/mocks/handlers/logout.handler.ts (hunk 1 above)
Comment (sahsisunny) [reply src/mocks/handlers/logout.handler.ts:10]: And when the API is failed?

Comment (anishpawaskar) [reply src/mocks/handlers/logout.handler.ts:10]: done

Comment (anishpawaskar) [reply src/mocks/handlers/logout.handler.ts:10]: > Can you share the API or screenshot?

sure will add it

**Pull request closed without merging**